### ISSUE TITLE
Stdlib::Fqdn does not allow leading _

### ISSUE THREAD

**Issue by tparkercbn** [puppetlabs/puppetlabs-stdlib#1400]
## Describe the Bug
Stdlib::Fqdn does not support leading _ characters in hostnames.  These are valid for most if not all SRV record for a domain.

eg: '_kerberos' 

## Expected Behavior
The leading _ should be allowed.

## Steps to Reproduce
Create a type using Stdlib::Fqdn and pass a value that begins with _ (ie: _kerberos)

## Environment
 - Stdlib 9.4

## Additional Context
None applicable

**Comment by tparkercbn** [comment]
I'm going to close this.  We need a new type for SRV records but I think it's actually correct for FQDN to reject the leading _ based on what is a valid domain name for everything except SRV.

**Comment by tparkercbn** [comment]
Closing

**Comment by ekohl** [comment]
I agree it shouldn't be in FQDN. There's `Stdlib::DNS::Zone`. I suppose somewhere in that namespace some SRV type would be the best place.
